### ISSUE TITLE
Differing behavior between Aead.[encrypt|decrypt] vs DeterministicAead.[encrypt|decrypt]Deterministically when passing null for additionalData

### ISSUE THREAD

**Issue by rayman42003** [google/tink#286]
I've been encrypting values with Google Tink successfully using AES256_GCM, and was passing in null for the additionalData parameter in encrypt/decrypt for Aead (i.e. `encrypt(clearBytes, null);`).

Some things have come up and now I need to use DeterministicAead (AES256_SIV).  Calling encrypt/decrypt Deterministically while passing in null parameter to additionalData (i.e. `encryptDeterministically(clearBytes, null)` results in a NullPointerException:

```
java.lang.NullPointerException
	at com.google.crypto.tink.subtle.AesCmac.computeMac(AesCmac.java:77)
	at com.google.crypto.tink.subtle.AesSiv.s2v(AesSiv.java:78)
	at com.google.crypto.tink.subtle.AesSiv.decryptDeterministically(AesSiv.java:139)
	at com.google.crypto.tink.daead.DeterministicAeadFactory$1.decryptDeterministically(DeterministicAeadFactory.java:100)
...
```

I think it has to do with the fact that `com.google.crypto.tink.subtle.AesSiv` has a method `private byte[] s2v(final byte[]... s)` that doesn't handle the null from [encrypt|decrypt]Deterministically for additionalData.

This is how it's handled in `com.google.crypto.tink.subtle.AesGcmJce`, for example:

```
if (associatedData != null && associatedData.length != 0) {
    cipher.updateAAD(associatedData);
}
```

The method works as expected when passing `new byte[0]` in for additionalData.

There's not really any contract from the methods that they should accept null values so it might not be actually considered a bug.  But, it was taking me some time to figure out why it wasn't possible to just replace Aead with DeterminisiticAead everywhere, and thought that having consistent behavior would be worth raising the issue.

**Comment by thaidn** [comment]
Thanks for reporting this bug.

Could you please send a PR to fix it? I think it's relatively simple and you've already figured out the root cause.

Thanks again!

**Comment by chuckx** [comment]
Fixed by #287.
